sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py

```diff
@@ -0,0 +1,216 @@
+        self._config = Configuration(**kwargs)
+        self._connection = None
+        self._entity_name = kwargs.get("entity_name")
+        self._auth_uri = "sb://{}".format(self.fully_qualified_namespace)
+        if self._entity_name:
+            self._auth_uri = "{}/{}".format(self._auth_uri, self._entity_name)
+        # Internal flag for switching whether to apply connection sharing, pending fix in uamqp library
+        self._connection_sharing = False
+
+    def __enter__(self):
+        if self._connection_sharing:
+            self._create_uamqp_connection()
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+
+    def _create_uamqp_connection(self):
+        auth = create_authentication(self)
+        self._connection = uamqp.Connection(
+            hostname=self.fully_qualified_namespace,
+            sasl=auth,
+            debug=self._config.logging_enable
+        )
+
+    def close(self):
+        # type: () -> None
+        """
+        Close down the ServiceBus client and the underlying connection.
+
+        :return: None
+        """
+        if self._connection_sharing:
+            self._connection.destroy()
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str,
+        **kwargs
+    ):
+        # type: (str, Any) -> ServiceBusClient
+        """
+        Create a ServiceBusClient from a connection string.
+
+        :param str conn_str: The connection string of a Service Bus.
+        :keyword str entity_name: Optional entity name, this can be the name of Queue or Topic.
+         It must be specified if the credential is for specific Queue or Topic.
+        :keyword bool logging_enable: Whether to output network trace logs to the logger. Default is `False`.
+        :keyword transport_type: The type of transport protocol that will be used for communicating with
+         the Service Bus service. Default is `TransportType.Amqp`.
+        :paramtype transport_type: ~azure.servicebus.TransportType
+        :keyword dict http_proxy: HTTP proxy settings. This must be a dictionary with the following
+         keys: `'proxy_hostname'` (str value) and `'proxy_port'` (int value).
+         Additionally the following keys may also be present: `'username', 'password'`.
+        :rtype: ~azure.servicebus.ServiceBusClient
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+                :start-after: [START create_sb_client_from_conn_str_sync]
+                :end-before: [END create_sb_client_from_conn_str_sync]
+                :language: python
+                :dedent: 4
+                :caption: Create a new instance of the ServiceBusClient from connection string.
+
+        """
+        host, policy, key, entity_in_conn_str = _parse_conn_str(conn_str)
+        return cls(
+            fully_qualified_namespace=host,
+            entity_name=entity_in_conn_str or kwargs.get("entity_name"),
```

Comment:
If you're not popping entity_name, and you provide it here + expanding kwargs, won't you run into a "got multiple values for keyword argument" exception?  

there are a few options here; kwargs.pop, or have it be a named arg so it doesn't percolate through kwargs.  I tend to prefer explicit arguments when possible, but would bounce this past @annatisch as well to ensure I'm not just letting my bias against kwargs break us into jail.

Reply:
sounds very reasonable to me!
I should apply the "pop" pattern

Reply:
updated!